### DISCUSSION TITLE
State that is always processed after current frame

### DISCUSSION THREAD

**Discussion by potsmugen** [ikemen-engine/Ikemen-GO#1187] [Feature Request]
### Is your feature request related to a problem? Please describe.

As discussed on and off on Discord. I'm posting it here to keep track of it.

One of the limitations of negative states is that they usually can only recognize that the character changed states in the next frame. As such you can't do any checks from them at time = 0.

### Describe the solution you'd like

The solution is simple: a state that is processed at the end of the current frame, rather than at the beginning like the negative states. An intuitive name for it could be State +1.

So characters would be processed like this:
State -4 > State -3 > State -2 > State -1 > Current State > State +1

This state could have the same properties as State -4 (ignore pauses, read by helpers) since it's easy to limit that freedom in ZSS if one wants to do so (if !pausetime && !ishelper && etc).
The alternative would be having states +4, +3, +2 and maybe +1, which is probably too much.

The more I think about this state the more I see ways it could improve some codes, so I'm sure it'd get a lot of mileage.

**Comment by ghost** [comment]
i recently found this out too and never thought it currently works this way, i was wondering why a noturntarget was still triggering a character to turn for 1 frame if i put it in a negative state. I ended up having to put it in the actual state itself outside of the negative states and it worked fine. very interesting

**Comment by potsmugen** [comment]
Implemented.
